vslib/src/SwitchStateBase.cpp

```diff
@@ -1811,6 +1815,17 @@ sai_status_t SwitchStateBase::refresh_read_only(
         }
     }
 
+    if (meta->objecttype == SAI_OBJECT_TYPE_SYSTEM_PORT)
+    {
+        switch (meta->attrid)
+        {
+            case SAI_SYSTEM_PORT_ATTR_TYPE:
+            case SAI_SYSTEM_PORT_ATTR_CONFIG_INFO:
+            case SAI_SYSTEM_PORT_ATTR_PORT:
+                return SAI_STATUS_SUCCESS;
```

Comment:
could SAI_SYSTEM_PORT_ATTR_PORT and SAI_SYSTEM_PORT_ATTR_TYPE ? do they need refresh ?

Reply:
No. These are read-only attributes and they are not modified after initialization time. No refresh is needed for these

Reply:
read_only attributes are considered that they can change value during switch runtime, for example port_number, where ports can be removed or added during switch runtime, and those are always recalculated

Reply:
Got it. Thanks for clarification. For a any given system port configuration (that is done only at the creation time), this is fixed depending on <switch_id, core index and core port index> - part of system port configuration. Since this is fixed through out the life of the system, we do not need refreshing of these attributes. However if the local port oid changes during run time, which unlikely, these need refresh

Reply:
local port oid is which object type ? SAI_OBJECT_TYPE_PORT ? or SAI_OBJECT_TYPE_SYSTEM_PORT?
since PORT can be removed/added during runtime when doing port split. if your voq information don't change during lifetime of switch object existence, regardless of any other operation, then you don't need to do any refreshing here

Reply:
Local port oid is SAI_OBJECT_TYPE_PORT. I read it from m_port_list to associate it with system port corresponding to the local ports (Refer PR #703). The system port config info tuple mentioned above will not change. If m_port_list is not changed during runt time, no need to do refresh here. But I see there is refresh for m_port_list in refresh_ports(). Does this mean previously assigned oid of the existing port will change? I am wondering, should I refresh system port config to associate system port with changed local port oid, if at all changed.

Reply:
object type port list can change during runtime, ports can be removed and added, i dont know details about voq and system ports, but seems to me that they are somehow related/dependent and it will be tricky to make this change and refresh operations

Reply:
Yes. I Agree. Without any processing existing for system port when local port is removed, it is unclear about having refreshing, especially since reading this attribute is one time during initialization of system ports. So for now, I'll not provide any refreshing for SAI_SYSTEM_PORT_ATTR_PORT. We'll re-visit this in the future.

Reply:
ok, just keep in mind that in case of warm boot scenario some wired things can happen, like duplicated port/system_ports etc and it can be hard to identify where this actually happens

Reply:
o.k. Thanks for the feedback.